internal/s3manager/s3manager.go

```diff
@@ -158,6 +158,21 @@ region = %s
 	return secret, err
 }
 
+// Stat retrieves the meta data for the blob specified by it blobUID.
+func (m *Manager) Meta(ctx context.Context, blobUID string) (map[string]string, error) {
+	if err := m.EnsureBucket(ctx); err != nil {
```

Comment:
This line will make a request to S3 every time. If the bucket is not there yet, it will create it but this method will fail later because obviously the object doesn't exist (since the bucket didn't exist either). We could as well let it fail because of the missing bucket. After all, it's a bit unexpected for this "getter" method to create the bucket. One can argue that it's also the case for the `Upload` method which does the same but at least there, the creation of the bucket is happening so that the method doesn't fail.

In other words, we can remove this call to `EnsureBucket` here.

Reply:
Ok. Will do. Will also look into writing a test.

Reply:
commit 189a36981220ee32960d620612dadaf51204995d

todo: test

Reply:
test commit 96a76c10c5f8d2ca795b8463549f51fb92bb6d78